ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml

```diff
@@ -67,100 +61,93 @@ Item {
             sourceModel: root.contactsModel
 
             function panelUsagePredicate(isVerified) {
-                if (panelUsage === Constants.contactsPanelUsage.verifiedMutualContacts) return isVerified
-                if (panelUsage === Constants.contactsPanelUsage.mutualContacts) return !isVerified
+                if (panelUsage === Constants.contactsPanelUsage.verifiedMutualContacts)
+                    return isVerified
+                if (panelUsage === Constants.contactsPanelUsage.mutualContacts)
+                    return !isVerified
+
                 return true
             }
 
-            function searchPredicate(name, pubkey) {
+            function searchPredicate(name, pubkey, compressedPubKey) {
                 const lowerCaseSearchString = root.searchString.toLowerCase()
-                const compressedPubkey = Utils.getCompressedPk(pubkey)
 
                 return name.toLowerCase().includes(lowerCaseSearchString) ||
                        pubkey.toLowerCase().includes(lowerCaseSearchString) ||
-                       compressedPubkey.toLowerCase().includes(lowerCaseSearchString)
+                       compressedPubKey.toLowerCase().includes(lowerCaseSearchString)
             }
 
             filters: [
-                ExpressionFilter { expression: filteredModel.panelUsagePredicate(model.isVerified) },
-                ExpressionFilter {
+                FastExpressionFilter {
+                    expression: filteredModel.panelUsagePredicate(model.isVerified)
+                    expectedRoles: ["isVerified"]
+                },
+                FastExpressionFilter {
```

Comment:
Could use the `SearchFilter` here?

Reply:
Yeah, could be but I don't want to overload this already big PR. Will do in some follow-ups.